src/app/core/services/phase.service.ts

```diff
@@ -98,7 +102,16 @@ export class PhaseService {
    * Retrieves the repository url from local storage and sets to current repository.
    */
   initializeCurrentRepository() {
-    const repo = new Repo(window.localStorage.getItem('org'), window.localStorage.getItem('dataRepo'));
+    const org = window.localStorage.getItem('org');
+    const repoName = window.localStorage.getItem('dataRepo');
+    this.logger.info(`Phase Service: received initial org (${org}) and initial name (${repoName})`);
+    let repo: Repo;
+    if (org === 'undefined' || repoName === 'undefined') {
```

Comment:
I do not think that you should check against the string `'undefined'`. You should check using `!org` and `!repoName`

Reply:
Originally I had a check for this to see if `org` or `repoName` were truthy. However, localStorage's implementation is pretty funky. In `session-selection.component` when `org` and `repoName` are initially set, if the value itself is undefined or even null, `window.localStorage::setItem` essentially sets a key-value pair of the key to the string value of the the item being set. So, even if `org` or `repoName` were indeed undefined, they would be be set to the string `'undefined'`, which is why the check is required. Pretty weird IMO.

Reply:
I think what we can do instead is probably do a `JSON::parse` on the item that is retrieved from localStorage, I'll try and see if that helps to convert the value to the actual `undefined` value instead of a string of `'undefined'`

Reply:
that didn't work either as `JSON::parse` now throws an error if there is no `[` or `{` at the start of the string

Reply:
@chunweii what if in `SessionSelectionComponent`, when calling `SessionSelectionComponent::setupSession`, we check whether the `repoOrg` or `repoName` is undefined before setting it in localStorage? Currently, we set the values in localStorage regardless of whether it is undefined or not, which localStorage is simply setting as a string. That way, when we check again for org or repo in `phaseService`, if it was not set it would return null. Then we can check and verify that either values are truthy. WDYT?

Reply:
@vigneshsankariyer1234567890 Yes I think that is a good idea